### PR TITLE
Compute graph support

### PR DESCRIPTION
**Pull request by mohammadfawaz**
Bump `essential-base` and handle the compute graph stuff.

The change is mostly documented in the code to a certain extent. Really the approach involves building the data flow graph (from variables to other variables and constraints) and then compile things in topological order. Also handle the memory hand off between nodes in a reasonable way that matches what `essential-check` expects. 

The change also improves `validation_tests` to not need to hard code contract and predicate addresses in the JSON solution objects.

A bunch of remaining things that would be nice to have: https://github.com/essential-contributions/pint/issues/1016

### CODE REVIEW

File: pintc/src/asm_gen/asm_builder.rs [hunk 1]
```diff
@@ -620,7 +585,13 @@ impl<'a> AsmBuilder<'a> {
...
             }
             BinaryOp::NotEqual => {
-                asm.push(Pred::Eq.into());
+                let type_size = lhs.get_ty(contract).size(handler, contract)?;
+                if type_size == 1 {
+                    asm.push(Pred::Eq.into());
+                } else {
+                    asm.push(Stack::Push(type_size as i64).into());
+                    asm.push(Pred::EqRange.into());
+                }
```
Comment (mohammadfawaz) [inline pintc/src/asm_gen/asm_builder.rs:594]: This was a bug.

File: pintc/src/asm_gen/asm_builder.rs [hunk 2]
```diff
@@ -757,46 +728,85 @@ impl<'a> AsmBuilder<'a> {
...
-                match self.compile_expr_pointer(handler, asm, &args[0], contract, pred)? {
-                    Location::State(next_state) => {
-                        // Remove the placeholder for index computation since it is not needed for
-                        // the `SizeOf` opcode.
-                        asm.push(Stack::Pop.into());
-
-                        asm.push(Stack::Push(next_state as i64).into()); // delta
-                        asm.push(Access::StateLen.into()); // Range length for State
+                // This whole implementation is quite hacky, particularly for `Location::Memory`.
+                // I expect this to change asap.
```
Comment (mohammadfawaz) [inline pintc/src/asm_gen/asm_builder.rs:739]: This is crap. All of `__size_of` needs to be reworked properly. There's an issue for that.

**Comment by mohammadfawaz** [comment]
This is ready but still requires some extra cleanup post merge. Keeping track of a few things here: https://github.com/essential-contributions/pint/issues/1016

**Comment by mohammadfawaz** [comment]
Really most of the changes are in tests. For asm gen tests, these have become a pain so I'm proposing a refactor [here](https://github.com/essential-contributions/pint/issues/1028), but that's not simple to do.

**Review by otrho — APPROVED** [review]
I'll need to study how it all works later but in general LGTM.

**Pull request merged**